ARE/middleware/src/main/java/eu/asterics/mw/are/BundleManager.java

```diff
@@ -132,16 +144,173 @@ public BundleManager(ModelValidator modelValidator) {
+                String bundleDescriptorAsXMLString=ResourceRegistry.getResourceContentAsString(bundleDescriptorURI.toURL().openStream());
+                Set<IComponentType> componentTypeSet = DefaultBundleModelParser.instance.parseModelAsXMLString(bundleDescriptorAsXMLString);
+                for(IComponentType componentType : componentTypeSet) {
+                    componentTypeIDToJarName.put(componentType.getID(),ResourceRegistry.toString(ResourceRegistry.getInstance().toRelative(jarNameURI)));
+                    ComponentRepository.instance.install(componentType);
+                }
+            }
+            logger.fine("Found the following componentTypeIds: "+componentTypeIDToJarName.keySet().toString());
+        }
+    }
 
-        installServices();
+    /**
+     * Checks one of the used cache files is misssing.
+     * @return
+     */
+    private boolean cacheFileMissing() {
+        logger.fine("Checking missing cache files");
+        if(!ResourceRegistry.resourceExists(LOADER_COMPONENTLIST_CACHE_FILE_URI) ||
+                !ResourceRegistry.resourceExists(COMPONENT_COLLECTION_CACHE_FILE_URI)) {
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Compares the given md5sum hex string and checks it for equality with a newly created md5 hash string of all plugin jars.
+     * @param md5Sum
+     * @return
+     * @throws URISyntaxException
+     * @throws IOException
+     */
+    private boolean bundleChangeDetected(String md5Sum) throws URISyntaxException, IOException {
+        logger.fine("Checking bundle change...");
+        if(md5Sum==null) {
+            logger.warning("Given MD5 sum is null --> bundle change: true");
+            return true;
+        }
+        
+        URI pluginJarsHashFileURI=ResourceRegistry.getInstance().getResource(CACHING_PLUGIN_JARS_HASH_TXT, RES_TYPE.TMP);
+        File pluginJarsHashFile=ResourceRegistry.toFile(pluginJarsHashFileURI);
+        if(!pluginJarsHashFile.exists()) {
+            logger.warning("pluginJars hash file does not exist --> bundle change: true");
+            return true;
+        }
+        try(BufferedReader reader=new BufferedReader(new FileReader(pluginJarsHashFile))) {
+            String storedMD5Sum=reader.readLine();
+            if(md5Sum.equals(storedMD5Sum)) {
+                logger.fine("bundle change: false");
+                return false;
+            }
+        }
+        
+        logger.fine("bundle change: true");
+        return true;
+    }
+
+    /**
+     * Stores the given MD5 hash hex string to a file.
+     * @param md5Sum
+     * @throws URISyntaxException
+     * @throws IOException
+     */
+    private void storeMD5Sum(String md5Sum) throws URISyntaxException, IOException {
+        logger.fine("Storing MD5 sum...");
+        URI pluginJarsHashFileURI=ResourceRegistry.getInstance().getResource(CACHING_PLUGIN_JARS_HASH_TXT, RES_TYPE.TMP);
```

Comment:
why not using `ResourceRegistry#storeResource(java.lang.String, java.lang.String, ResourceRegistry.RES_TYPE)` here?

Reply:
ok